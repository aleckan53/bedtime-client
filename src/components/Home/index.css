@media(min-width: 850px) {
  .Home {
    padding: 0 20%
  }
}
@media(min-width: 1150px) {
  .Home {
    display: grid;
    padding: 0 20%;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 7rem;
  }
}

@media(min-width: 1430px) {
  .Home {
    display: grid;
    padding: 0 25%;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 7rem;
  }
}