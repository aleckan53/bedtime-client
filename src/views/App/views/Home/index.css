@media(min-width: 850px) {
  .Home {
    padding: 0 10%
  }
}
@media(min-width: 1100px) {
  .Home {
    padding: 0 20%
  }
}
@media(min-width: 1250px) {
  .Home {
    display: grid;
    padding: 0;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 2rem;
  }
}