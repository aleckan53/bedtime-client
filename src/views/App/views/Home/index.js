import React from 'react'
import styles from './styles.module.css'
import Stack from './components/Stack2'

const Home = props => {

  return (
    <section className={styles.Home}>
      <Stack data={props.stories} title='Top Rated'/>
      <Stack data={props.stories} title='New release'/>
    </section>
  )
}

export default Home