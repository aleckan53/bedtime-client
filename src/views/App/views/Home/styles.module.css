.Home {
  padding-top: 2rem;
}