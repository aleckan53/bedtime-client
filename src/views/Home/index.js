import React from 'react'
import Story from './components/Story'
import Stack from './components/Stack'
import { Switch, Route } from 'react-router-dom'

const Home = props => {
  return (
    <section>
      <Switch>
        <Route
          path={`/home/story/:id`}
          render={(routerProps) => <Story data={props.stories} {...routerProps}/>}
        />
        <Route 
          exact
          path={`${props.match.path}/`}
          render={() => <Stack data={props.stories}/>}
        />
      </Switch>
    </section>
  )
}

export default Home