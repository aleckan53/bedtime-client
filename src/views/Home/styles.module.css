.Home {
  padding-top: 3rem;
}