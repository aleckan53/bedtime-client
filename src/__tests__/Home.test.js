import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from '../components/Home'
import { BrowserRouter } from 'react-router-dom'

describe(`Home Component`, () => {
  const stories = [
    {id: 1, name: 'test', rating: 5},
    {id: 12, name: 'test2', rating: 3},
  ]

  it('renders without errors wit given props', () => {

    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Home stories={stories}/>
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(<Home stories={stories}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})