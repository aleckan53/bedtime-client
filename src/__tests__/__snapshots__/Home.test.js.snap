// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches the snapshot 1`] = `
<section
  className="Home"
>
  <Stack
    data={
      Array [
        Object {
          "id": 1,
          "name": "test",
          "rating": 5,
        },
      ]
    }
    rating={4}
    title="Top Rated"
  />
  <Stack
    data={
      Array [
        Object {
          "id": 12,
          "name": "test2",
          "rating": 3,
        },
      ]
    }
    rating={0}
    title="New release"
  />
</section>
`;
